src/main/java/io/confluent/connect/elasticsearch/Validator.java

```diff
@@ -238,7 +272,91 @@ private void validateSsl() {
     }
   }
 
+  private void validateConnection() {
+    try {
+      if (!client.ping(RequestOptions.DEFAULT)) {
+        throw new IOException("Cannot connect to Elasticsearch.");
+      }
+    } catch (IOException e) {
+      String errorMessage = String.format(
+          "Could not connect to Elasticsearch due to %s",
+          e.getMessage()
+      );
+      addErrorMessage(CONNECTION_URL_CONFIG, errorMessage);
+
+      if (config.isAuthenticatedConnection()) {
+        errorMessage = String.format(
+            "Could not authenticate the user. Check the '%s' and '%s'. Error message: %s.",
+            CONNECTION_USERNAME_CONFIG,
+            CONNECTION_PASSWORD_CONFIG,
+            e.getMessage()
+        );
+        addErrorMessage(CONNECTION_USERNAME_CONFIG, errorMessage);
+        addErrorMessage(CONNECTION_PASSWORD_CONFIG, errorMessage);
+      }
+
+      if (config.isSslEnabled()) {
+        errorMessage = String.format(
+            "Could not connect to Elasticsearch. Check your SSL settings. Error message: %s",
+            e.getMessage()
+        );
+
+        addErrorMessage(SECURITY_PROTOCOL_CONFIG, errorMessage);
+      }
+
+      if (config.isKerberosEnabled()) {
+        errorMessage = String.format(
+            "Could not connect to Elasticsearch. Check your Kerberos settings. Error message: %s",
```

Comment:
nit: same as for SSL error, can we give more guidance to the user, which configs should she check?

Reply:
same comment as above, this is added to every kerberos config. hence resolving